pxr/usdValidation/usdUtilsValidators/testenv/testUsdUtilsValidators.cpp

```diff
@@ -113,11 +126,102 @@ TestPackageEncapsulationValidator()
     TF_AXIOM(errors.empty());
 }
 
+static void
+TestMissingReferenceValidator()
+{
+    UsdValidationRegistry& registry = UsdValidationRegistry::GetInstance();
+
+    // Verify the validator exists
+    const UsdValidationValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdUtilsValidatorNameTokens->missingReferenceValidator);
+
+    TF_AXIOM(validator);
+
+    const UsdStageRefPtr& stage = UsdStage::CreateInMemory();
+
+    const UsdGeomXform xform = UsdGeomXform::Define(stage, SdfPath("/Xform"));
+    const SdfReference badLayerReference("doesNotExist.usd");
+    xform.GetPrim().GetReferences().AddReference(badLayerReference);
+
+    // Validate an error occurs on a missing layer reference
+    {
+        const UsdValidationErrorVector errors = validator->Validate(stage);
+
+        // Verify both the layer errors are present
+        const std::string expectedErrorMessage = "Found unresolvable external "
+                                                 "dependency 'doesNotExist.usd'.";
+        const TfToken expectedIdentifier =
+            TfToken(
+            "usdUtilsValidators:MissingReferenceValidator.UnresolvableDependency");
+        TF_AXIOM(errors.size() == 1);
+
+        ValidateError(errors[0], expectedErrorMessage,
+                expectedIdentifier);
+    }
+
+    xform.GetPrim().GetReferences().RemoveReference(badLayerReference);
+    const SdfReference badAssetReference("doesNotExist.jpg");
+    xform.GetPrim().GetReferences().AddReference(badAssetReference);
+
+    // Validate an error occurs on a missing asset reference
+    {
+        const UsdValidationErrorVector errors = validator->Validate(stage);
+
+        // Verify both the layer errors are present
+        const std::string expectedErrorMessage = "Found unresolvable external "
+                                                 "dependency 'doesNotExist.jpg'.";
+        const TfToken expectedIdentifier =
+            TfToken(
+            "usdUtilsValidators:MissingReferenceValidator.UnresolvableDependency");
+        TF_AXIOM(errors.size() == 1);
+
+        ValidateError(errors[0], expectedErrorMessage,
+                expectedIdentifier);
+    }
+
+    xform.GetPrim().GetReferences().RemoveReference(badAssetReference);
+    // Set an unresolved dependency on an unused variant
+
+    UsdVariantSets variantSets = xform.GetPrim().GetVariantSets();
+    UsdVariantSet colorsVariantSet = variantSets.AddVariantSet("myVariant");
+
+    UsdGeomXform selectedXform = UsdGeomXform::Define(stage, xform.GetPath().AppendChild(TfToken("selected")));
+    UsdGeomXform notSelectedXform = UsdGeomXform::Define(stage, xform.GetPath().AppendChild(TfToken("notSelected")));
```

Comment:
Don't think this is correct.

You can refer the authoring variant tutorials here: https://openusd.org/docs/Authoring-Variants.html

to see how you can add a variantSet with 2 variants on a prim, with one having a  good reference and other a bad reference. Have the one which has a good reference be the default set on the prim.

Does that make sense?

Reply:
Yep, thanks for the help here. Should be good now